### PR TITLE
BOM-1240

### PR DESCRIPTION
**Pull request by awais786**
Adding makefile.

### CODE REVIEW

**Comment by awais786** [comment]
```
RROR: test_updating_to_same_active_start_date (openedxstats.apps.sites.tests.UpdateSiteTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arbisoft/Documents/devstack/openedxstats/openedxstats/apps/sites/views.py", line 255, in add_site
    new_site.validate_unique()
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/db/models/base.py", line 937, in validate_unique
    raise ValidationError(errors)
django.core.exceptions.ValidationError: {'__all__': ['Site with this Url and Active start date already exists.']}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/arbisoft/Documents/devstack/openedxstats/openedxstats/apps/sites/tests.py", line 502, in test_updating_to_same_active_start_date
    response = self.client.post('/sites/update_site/'+str(new_site.pk)+'/', form_data, follow=True)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/test/client.py", line 515, in post
    secure=secure, **extra)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/test/client.py", line 314, in post
    secure=secure, **extra)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/test/client.py", line 380, in generic
    return self.request(**r)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/test/client.py", line 467, in request
    six.reraise(*exc_info)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/utils/six.py", line 686, in reraise
    raise value
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/arbisoft/Documents/devstack/openedxstats/openedxstats/apps/sites/views.py", line 258, in add_site
    return render_site_form(request, form, pk)
  File "/Users/arbisoft/Documents/devstack/openedxstats/openedxstats/apps/sites/views.py", line 377, in render_site_form
    {'form': form, 'post_url': reverse('sites:update_site', args=[pk]), 'page_title': 'Update Site'})
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/shortcuts.py", line 67, in render
    template_name, context, request=request, using=using)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/loader.py", line 97, in render_to_string
    return template.render(context, request)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/backends/django.py", line 95, in render
    return self.template.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 206, in render
    return self._render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/loader_tags.py", line 69, in render
    result = block.nodelist.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/templatetags/static.py", line 103, in render
    url = self.url(context)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/contrib/staticfiles/templatetags/staticfiles.py", line 16, in url
    return static(path)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/contrib/staticfiles/templatetags/staticfiles.py", line 9, in static
    return staticfiles_storage.url(path)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/contrib/staticfiles/storage.py", line 131, in url
    hashed_name = self.stored_name(clean_name)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/contrib/staticfiles/storage.py", line 280, in stored_name
    cache_name = self.clean_name(self.hashed_name(name))
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/django/contrib/staticfiles/storage.py", line 94, in hashed_name
    (clean_name, self))
ValueError: The file 'sites/favicon.ico' could not be found with <whitenoise.storage.CompressedManifestStaticFilesStorage object at 0x10e3de208>.
```

**Comment by awais786** [comment]
```
FAIL: test_add_a_new_version (openedxstats.apps.sites.tests.SubmitSiteFormTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arbisoft/Documents/devstack/openedxstats/openedxstats/apps/sites/tests.py", line 164, in test_add_a_new_version
    self.assertEqual(updated_site.course_type, 'Both')
AssertionError: 'Unknown' != 'Both'
- Unknown
+ Both


======================================================================
FAIL: test_updating_with_valid_changes (openedxstats.apps.sites.tests.UpdateSiteTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arbisoft/Documents/devstack/openedxstats/openedxstats/apps/sites/tests.py", line 480, in test_updating_with_valid_changes
    self.assertEqual(updated_site.course_type, 'MOOC')
AssertionError: 'SPOC' != 'MOOC'
- SPOC
+ MOOC
```

**Comment by awais786** [comment]
FAILED (failures=2, errors=13)

**Comment by awais786** [comment]
```
======================================================================
ERROR: test_todays_logs (openedxstats.apps.sites.tests.ReferrerLogTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arbisoft/Documents/devstack/openedxstats/openedxstats/apps/sites/tests.py", line 555, in setUp
    self.conn = boto.connect_s3()
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/boto/__init__.py", line 141, in connect_s3
    return S3Connection(aws_access_key_id, aws_secret_access_key, **kwargs)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/boto/s3/connection.py", line 191, in __init__
    validate_certs=validate_certs, profile_name=profile_name)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/boto/connection.py", line 569, in __init__
    host, config, self.provider, self._required_auth_capability())
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/boto/auth.py", line 991, in get_auth_handler
    'Check your credentials' % (len(names), str(names)))
boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials

**Comment by awais786** [comment]
@jmbowman few tests are failing on master and also boto errors appearing.

File: openedxstats/apps/sites/views.py [hunk 1]
```diff
@@ -203,10 +203,15 @@ def generate_summary_data(self, start_datetime):
...
+
+            daily_courses_count = day_stats['courses']
+            if daily_courses_count is None:
+                daily_courses_count = 0
+
             daily_summary_obj = SiteSummarySnapshot(
                 timestamp=day,
                 num_sites=day_stats['sites'],
-                num_courses=day_stats['courses'] - over_count,
+                num_courses=daily_courses_count - over_count,
```
Comment (awais786) [inline openedxstats/apps/sites/views.py]: None-0 was throwing error.

**Comment by awais786** [comment]
@jmbowman I think in this [commit](https://github.com/edx/openedxstats/commit/4ee40ed06013426be7f130d0a95a82e75ea978a8)
they have removed few fields from form but never update the tests.

**Comment by awais786** [comment]
@jmbowman for these [tests](https://github.com/edx/openedxstats/blob/master/openedxstats/apps/sites/tests.py#L539) they need valid credentials.
Maybe we can skip them ?

File: openedxstats/apps/sites/tests.py [hunk 2]
```diff
@@ -161,7 +165,7 @@ def test_add_a_new_version(self):
         old_site = Site.objects.filter(url='https://test.com').order_by('-active_start_date').last()
         self.assertEqual(Site.objects.count(), 2)
         self.assertEqual(Site.objects.filter(url='https://test.com').count(), 2)
-        self.assertEqual(updated_site.course_type, 'Both')
+        self.assertEqual(updated_site.course_type, 'Unknown')
```
Comment (awais786) [inline openedxstats/apps/sites/tests.py]: form does not have course_type field and thats why picking default value.

File: openedxstats/settings/testing.py [hunk 3]
```diff
@@ -1,3 +1,15 @@
 from .base import *  # noqa
 
 # By default, nothing is different for testing.
+
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
```
Comment (jmbowman) [inline openedxstats/settings/testing.py]: The README implied that postgres is optional for tests; if that's inaccurate, we should either fix the tests or update the README.

**Comment by awais786** [comment]
@jmbowman Please review it. My 2nd Pr in progress with make upgrade stuff.

**Pull request merged**